File: DakNong_2324/Bai3/FILEDEL.cpp
#include <iostream>
#include <fstream>
#include <unordered_set>
#include <vector>

using namespace std;

int main() {
    ifstream inputFile("FILEDEL.INP");
    ofstream outputFile("FILEDEL.OUT");

    int n, q;
    inputFile >> n >> q;
    
    vector<string> files(n);
    vector<int> remaining_files;
    unordered_set<char> deleted_chars;
    
    // Đọc danh sách tệp tin
    for (int i = 0; i < n; ++i) {
        inputFile >> files[i];
    }

    // Xử lý từng thao tác xóa
    for (int i = 0; i < q; ++i) {
        char delete_char;
        inputFile >> delete_char;

        deleted_chars.insert(delete_char);
        int count = 0;

        for (const auto& file : files) {
            bool to_delete = false;

            for (char ch : file) {
                if (deleted_chars.count(ch)) {
                    to_delete = true;
                    break;
                }
            }
            
            if (!to_delete) {
                count++;
            }
        }

        // Lưu số lượng tệp tin còn lại sau mỗi thao tác xóa
        remaining_files.push_back(count);
    }

    // Ghi kết quả ra file
    for (int i = 0; i < q; ++i) {
        outputFile << remaining_files[i] << endl;
    }

    inputFile.close();
    outputFile.close();
    
    return 0;
}

File: THCS/2023_2024/DakNong_2324/Bai3/FILEDEL.py
def main():
    with open("FILEDEL.INP", "r") as inputFile:
        # Đọc số lượng tệp tin và số thao tác xóa
        n, q = map(int, inputFile.readline().split())
        
        # Đọc tên các tệp tin
        files = [inputFile.readline().strip() for _ in range(n)]
        
        # Đọc các ký tự để xóa
        delete_chars = [inputFile.readline().strip() for _ in range(q)]
    
    deleted_chars = set()
    remaining_files = []
    
    # Xử lý từng thao tác xóa
    for delete_char in delete_chars:
        deleted_chars.add(delete_char)
        count = 0
        
        # Kiểm tra từng tệp tin
        for file in files:
            to_delete = False
            
            # Kiểm tra xem có ký tự nào trong tệp tin thuộc tập deleted_chars
            for ch in file:
                if ch in deleted_chars:
                    to_delete = True
                    break
            
            # Nếu tệp tin không chứa ký tự nào cần xóa, đếm lại
            if not to_delete:
                count += 1
        
        # Lưu số lượng tệp tin còn lại sau mỗi thao tác xóa
        remaining_files.append(count)
    
    # Ghi kết quả vào file
    with open("FILEDEL.OUT", "w") as outputFile:
        for count in remaining_files:
            outputFile.write(f"{count}\n")

if __name__ == "__main__":
    main()